packages/pn-personagiuridica-webapp/src/components/IntegrazioneApi/PublicKeysTable.tsx

```diff
@@ -0,0 +1,145 @@
+import { add } from 'date-fns';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+  EmptyState,
+  KnownSentiment,
+  Row,
+  SmartBody,
+  SmartBodyCell,
+  SmartBodyRow,
+  SmartHeader,
+  SmartHeaderCell,
+  SmartTable,
+  SmartTableData,
+  formatDate,
+} from '@pagopa-pn/pn-commons';
+
+import { BffPublicKeysResponse, PublicKeyRow } from '../../generated-client/pg-apikeys';
+import { ApiKeyColumnData, ModalApiKeyView } from '../../models/ApiKeys';
+import ApiKeysDataSwitch from './ApiKeysDataSwitch';
+
+type Props = {
+  publicKeys: BffPublicKeysResponse;
+  handleModalClick: (view: ModalApiKeyView, publicKeyId: string) => void;
+};
+
+const PublicKeysTable: React.FC<Props> = ({ publicKeys, handleModalClick }) => {
+  const { t } = useTranslation(['integrazioneApi']);
+
+  const data: Array<Row<ApiKeyColumnData>> = publicKeys.items.map((n: PublicKeyRow) => ({
+    ...n,
+    date: formatDate(add(new Date(n.createdAt ?? ''), { days: 355 }).toISOString()),
```

Comment:
I don't understand why we have to add 355 days

Reply:
This is specified in the task: https://pagopa.atlassian.net/issues/PN-12793?filter=11328

> scadenza della chiave (calcolata partendo dalla creation date della chiave + 355 giorni)

Reply:
Ok pay attention to `new Date(n.createdAt ?? '')`. If n.createdAt is undefined and we use '', this will cause an error because `new Date('')` is an invalid date and add something to an invalid date leads to an error.
So we can do something like that `n.createdAt ? formatDate(add(new Date(n.createdAt)...: ''`